asmcomp/arm64/emit.mlp

```diff
@@ -741,11 +741,13 @@ let emit_instr env i =
           `	add	{emit_reg reg_stack_arg_end}, sp, #{emit_int (Misc.align stack_ofs 16)}\n`;
           emit_load_symbol_addr reg_x8 func;
           `	bl	{emit_symbol "caml_c_call_stack_args"}\n`;
-          `{record_frame env i.live (Dbg_other i.dbg)}\n`
+          `{record_frame env i.live (Dbg_other i.dbg)}\n`;
+          `	dmb	ishld\n`
         end else if alloc then begin
           emit_load_symbol_addr reg_x8 func;
           `	bl	{emit_symbol "caml_c_call"}\n`;
-          `{record_frame env i.live (Dbg_other i.dbg)}\n`
+          `{record_frame env i.live (Dbg_other i.dbg)}\n`;
+          `	dmb	ishld\n`
```

Comment:
This instruction could be part of `caml_c_call` instead, saving a small amount of code size.

Also, the C-to-OCaml transition in `caml_start_program` (during startup and `caml_callback`) needs the same instruction

Reply:
Fixed in https://github.com/ocaml/ocaml/pull/13393/commits/1e99015f2cf91980a0b906c10dc97c30d364db76.